client/packages/react/src/useQuery.ts

```diff
@@ -37,10 +47,9 @@ export function useQuery<
   // uses `Object.is` to compare the previous and next state.
   // If we don't use a ref, the state will always be considered different, so
   // the component will always re-render.
-  const resultCacheRef =
-    useRef<LifecycleSubscriptionState<Q, Schema, WithCardinalityInference>>(
-      defaultState,
-    );
+  const resultCacheRef = useRef<
+    LifecycleSubscriptionState<Q, Schema, WithCardinalityInference>
```

Comment:
Hmm...I am seeing some caching issues when I play with some data in Explorer. I will look a bit more deeply